tutorials/international/index.md

```diff
@@ -0,0 +1,281 @@
+it. This happens because Localizations works like an
+[`InheritedWidget`](https://docs.flutter.io/flutter/widgets/InheritedWidget-class.html). When
+a build function refers to an inherited widget an implicit dependency
+on the inherited widget is created. When an inherited widget changes
+(when the Localizations widget's locale changes), its dependent
+contexts are rebuilt.
+
+Localized values are loaded by the Localizations widget's list of
+[`LocalizationsDelegate`](https://docs.flutter.io/flutter/widgets/LocalizationsDelegate-class.html)s. Each
+delegate must define an async
+[`load()`](https://docs.flutter.io/flutter/widgets/LocalizationsDelegate/load.html)
+method that produces an object which encapsulates a collection of
+localized values. Typically these objects define one method per localized value.
+
+In a large app, different modules or packages might be bundled with
+their own localizations. That's why the Localizations widget manages a
+table of objects, one per LocalizationsDelegate. To retrieve the
+object produced by one of the LocalizationsDelegate's `load` methods,
+you specify a BuildContext and the object's type.
+
+For example the localized strings for the Material widgets are defined
+by the
+[`MaterialLocalizations`](https://docs.flutter.io/flutter/material/MaterialLocalizations-class.html)
+class. Instances of this class are created by a LocalizationDelegate
+provided by the
+[`MaterialApp`](https://docs.flutter.io/flutter/material/MaterialApp-class.html)
+ class. They can be retrieved with `Localizations.of`:
+
+```dart
+Localizations.of<MaterialLocalizations>(context, MaterialLocalizations);
+```
+
+This particular `Localizations.of()` expression is used frequently, so the MaterialLocalizations class provides a convenient shortand:
+
+```dart
+static MaterialLocalizations of(BuildContext context) {
+  return Localizations.of<MaterialLocalizations>(context, MaterialLocalizations);
+}
+
+/// References to the localized values defined by MaterialLocalizations
+/// are typically written like this:
+
+tooltip: MaterialLocalizations.of(context).backButtonTooltip,
+```
+
+### A Class for the App's Localized Resources
+
+Putting all of this together for an internationalized app usually
+starts with the class that encapsulates the app's localized
+values. The example that follows is typical of such classes. The
+`MyLocalizations` class contains the app's strings (just one for the example)
+translated into the locales that the app supports. It uses the `initializeMessages()` function
+generated by the [Dart `intl` package](https://pub.dartlang.org/packages/intl)
+to load the translated strings and
+[`Intl.message()`](https://www.dartdocs.org/documentation/intl/0.15.1/intl/Intl/message.html)
+to look them up.
+
+```dart
+class MyLocalizations {
+  MyLocalizations(this.locale);
+
+  final Locale locale;
+
+  static Future<MyLocalizations> load(Locale locale) {
+    return initializeMessages(locale.toString())
```

Comment:
could we use async/await here?

Reply:
I'd rather not make this method async

Reply:
no problem, as long as we had a reason not to use async. When all things are equal, async/await sells Dart more.